src/os/bluestore/BlueFS.cc

```diff
@@ -479,6 +479,29 @@ void BlueFS::_update_logger_stats()
   }
 }
 
+void BlueFS::collect_read_stalled_alerts(osd_alert_list_t& alerts) {
+  if (!cct->_conf->bluestore_stalled_read_event_counter_threshold) {
+    return;
+  }
+  for (size_t i = 0; i < bdev.size(); ++i) {
+    if (!bdev[i]) continue;
+    bdev[i]->trim_stalled_read_event_queue(mono_clock::now());
+    size_t qsize = bdev[i]->get_stalled_read_event_queue_size();
+    if (qsize) {
```

Comment:
And even more generally - IMO it's better to encapsulate this alert creation into BlockDevice. This way we have the config parameters usage at a single place and bluefs would be almost unaware of this functionality.